src/ensRegistry.ts

```diff
@@ -83,7 +120,9 @@ function _handleNewOwner(event: NewOwnerEvent, isMigrated: boolean): void {
   domain.parent = event.params.node.toHexString()
   domain.labelhash = event.params.label
   domain.isMigrated = isMigrated
-  domain.save()
+  if (recurseDomainDelete(domain) === domain.id) {
```

Comment:
Can this be moved into the `recurseDomainDelete` function, so you're not having to do it repeatedly?